internal/controller/api/connection_mediator_v2_test.go

```diff
@@ -0,0 +1,108 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/RedHatInsights/cloud-connector/internal/config"
+	"github.com/RedHatInsights/cloud-connector/internal/domain"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	MESSAGE_ENDPOINT_V2 = URL_BASE_PATH + "/v2/connections/345/message"
+	ACCOUNT_NUMBER      = domain.AccountID("1234")
+)
+
+func MockedGetConnectionByClientID(context.Context, *logrus.Entry, domain.OrgID, domain.ClientID) (domain.ConnectorClientState, error) {
+	return domain.ConnectorClientState{Account: ACCOUNT_NUMBER, ClientID: "345"}, nil
+}
+
+func MockedGetConnectionsByOrgID(context.Context, *logrus.Entry, domain.OrgID, int, int) (map[domain.ClientID]domain.ConnectorClientState, int, error) {
+	return map[domain.ClientID]domain.ConnectorClientState{"345": {Account: ACCOUNT_NUMBER, ClientID: "345"}}, 1, nil
+}
+
+var _ = Describe("ConnectionMediatorV2", func() {
+
+	var (
+		cm                  *ConnectionMediatorV2
+		validIdentityHeader string
+	)
+
+	BeforeEach(func() {
+		apiMux := mux.NewRouter()
+		cfg := config.GetConfig()
+
+		proxyFactory := &MockClientProxyFactory{}
+
+		cm = NewConnectionMediatorV2(MockedGetConnectionByClientID, MockedGetConnectionsByOrgID, proxyFactory, apiMux, URL_BASE_PATH, cfg)
+		cm.Routes()
+
+		validIdentityHeader = buildIdentityHeader(ACCOUNT_NUMBER, "Associate")
```

Comment:
The `buildIdentityHeader()` method really needs to be modified to take in the org-id but we do not have to make that change now.  We should probably make that change in a different pr.

Reply:
Agree, I needed to match the org id I used for the mock functions with the one that's inside `buildIdentityHeader()` :+1: